tests/test_main.py

```diff
@@ -0,0 +1,146 @@
+import pytest
+from unittest.mock import MagicMock
+
+from src.wordle_manager.main import main
+from src.wordle_manager.utils import WordListManager
+from src.wordle_manager import words
+
+
+@pytest.fixture(autouse=True)
+def safe_testing_environment():
+    """Fixture to prevent any file writes and preserve original word list"""
+    original_words = words.word_list.copy()
+    original_id = id(words.word_list)
+
+    with pytest.MonkeyPatch.context() as monkeypatch:
```

Comment:
is this setup / patching needed? 
can we just init the class with an isolated test word list?

```
class WordListManager:
    def __init__(
        self, word_list: list[str] | None = None, save_on_change: bool = True
    ) -> None:
```